File: api/internal/data/models/flashcard.go
package models

import "time"

type Flashcard struct {
	ID        int32     `json:"id"`
	Name      string    `json:"name"`
	Author    string    `json:"author"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
	Color     string    `json:"color"`
	WordQty   int       `json:"wordQty"`
}

type CreateFlashcardRequest struct {
	FlashcardName  string `json:"flashcardName" validate:"required,min=1,max=64" mod:"trim"`
	FlashcardColor string `json:"flashcardColor" validate:"required,min=7,max=9"`
	WordsIds       []int  `json:"wordsIds"`
}

type GetFlashcardByIdResponse struct {
	Flashcard
	Words []Word `json:"words"`
}

type AssignFlashcardWordsRequest struct {
	WordsIds []int `json:"wordsIds"`
}

File: mobile/core/api/models/flashcard.ts
import { DictionaryWord } from "@/core/api/models/dictionary";

export interface FlashcardBrief {
  id: number;
  name: string;
  author: string;
  createdAt: Date;
  updatedAt: Date;
  color: string;
  wordQty: number;
}

export interface FlashcardDetails extends FlashcardBrief {
  words: DictionaryWord[];
}

export interface CreateFlashcardRequest {
  flashcardName: string;
  flashcardColor: string;
  wordsIds: number[];
}
export interface CreateFlashcardResponse extends FlashcardBrief {}
export interface FlashcardDetailsResponse extends FlashcardBrief {
  words: DictionaryWord[];
}

export interface UpdateFlashcardRequest extends CreateFlashcardRequest {
  id: number;
}
export interface UpdateFlashcardResponse extends CreateFlashcardResponse {}
